### PR TITLE
add data source for nomad ACL tokens

### LINKED ISSUE #85: Create data source for ACL token
# nomad_acl_token

Get information on an [ACL token](https://www.nomadproject.io/api/acl-tokens.html#read-token). 

**Important:** this data source will store tokens in the Terraform state. The documentation should have a warning like the [one in the resource page](https://github.com/terraform-providers/terraform-provider-nomad/blob/master/website/docs/r/acl_token.html.markdown).

## Example usage

```hcl
data "nomad_acl_token" "my_token" {
  accessor_id = "aa534e09-6a07-0a45-2295-a7f77063d429"
}
```

## Argument reference

* `accessor_id`: `(string)` Non-sensitive identifier for this token.

## Attributes reference

* `name`: `(string)` Human-friendly name of the ACL token.
* `accessor_id`: `(string)` Non-sensitive identifier for this token.
* `secret_id`: `(string)` The token value itself.
* `type`: `(string)` The type of the token.
* `policies`: `(list of strings)` List of policy names associated with this token.
* `global`: `(bool)` Whether the token is replicated to all regions, or if it will only be used in the region it was created.
* `create_time`: `(string)` Date and time the token was created.

### PR DESCRIPTION
**Pull request by softpunks**
This PR adds the data source for ACL tokens resource.

closes #85

### CODE REVIEW

File: website/docs/d/acl_token.html.markdown [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+layout: "nomad"
+page_title: "Nomad: nomad_acl_token"
+sidebar_current: "docs-nomad-datasource-acl-token"
+description: |-
+  Get information on an ACL token.
+---
+
+# nomad_acl_token
+
+Get information on an ACL token
```
Comment (lgfa29) [inline website/docs/d/acl_token.html.markdown]: Add a period at the end of the sentence.

File: website/nomad.erb [hunk 2]
```diff
@@ -25,6 +25,9 @@
             <li<%= sidebar_current("docs-nomad-datasource-regions") %>>
               <a href="/docs/providers/nomad/d/regions.html">nomad_regions</a>
             </li>
+            <li<%= sidebar_current("docs-nomad-datasource-acl-token") %>>
+              <a href="/docs/providers/nomad/d/acl_token.html">nomad_acl_token</a>
+          </li>
```
Comment (lgfa29) [inline website/nomad.erb]: Move this item to the top of the list to keep it alphabetically sorted.

File: nomad/data_source_acl_token.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	// retrieve the token
+	log.Printf("[DEBUG] Reading ACL Token %q", accessor)
+	token, _, err := client.ACLTokens().Info(accessor, nil)
+	if err != nil {
+
+		// As of Nomad 0.4.1, the API client returns an error for 404
+		// rather than a nil result, so we must check this way.
+		if strings.Contains(err.Error(), "404") {
+			d.SetId("")
+			return err
```
Comment (lgfa29) [inline nomad/data_source_acl_token.go]: You don't need to return `err` in this case, you can just `return nil` since this is not a failure.

**Comment by softpunks** [comment]
@cgbaker I added the test for `create_time` but I can't seem to get it to pass. Not sure if the problem is a) a bug in my test or b) `create_time` isn't getting set in the course of the test for some reason. Could anyone point me in the right direction on this?

**Pull request merged**